### DISCUSSION TITLE
Is Google ORTools supposed to be 600 megabytes when built?

### DISCUSSION THREAD

**Discussion by CeaseHarping** [google/or-tools#2949] [Build and configuration issues] [answered]
If I make a new C# project, add Google.OrTools 9.1.9490 from nuget, add a bit of code that uses it and build the project the output folder is about 600 megabytes, mostly ortools.lib:

![image](https://user-images.githubusercontent.com/95501974/144663034-26284596-3a75-492f-942d-14329eea122f.png)

Is this correct? Any way I can pare that down to , say, less than 10mb? I need just one function (Min Cost Flow) from the library..

**Comment by Mizux** [comment, marked as answer]
Yes an No,

Yes ortools.lib is such size but no it shouldn't be part of the .nupkg (shame on me)
Should be fixed in the next release coming soon.

ps: duplicate of https://github.com/google/or-tools/discussions/2800

**Reply by Mizux** [reply]
should be fixed by https://github.com/google/or-tools/commit/a7823594e1d9a28be0afce3e9e772a79b1b1f740
and CMake based build should already work
https://github.com/google/or-tools/blob/a7823594e1d9a28be0afce3e9e772a79b1b1f740/ortools/dotnet/Google.OrTools.runtime.csproj.in#L27

thanks to the test `$<STREQUAL:$<TARGET_PROPERTY:@PROJECT_NAME@,TYPE>,SHARED_LIBRARY>`
I'll monitor CI tomorrow...

**Reply by CaiusJard** [reply]
Hi Mizux, is there a date pencilled in for release of the next version?

**Reply by Mizux** [reply]
hope to be today: if I manage to generate all artifacts...
